applications/FluidDynamicsApplication/custom_elements/d_vms_dem_coupled.cpp

```diff
@@ -0,0 +1,497 @@
+// Protected functions
+
+///////////////////////////////////////////////////////////////////////////////////////////////////
+// Evaluation of system terms on Gauss Points
+
+template< class TElementData >
+void DVMSDEMCoupled<TElementData>::AddVelocitySystem(
+    TElementData& rData,
+    MatrixType &rLocalLHS,
+    VectorType &rLocalRHS)
+{
+    auto& LHS = rData.LHS;
+    LHS.clear();
+
+    const double density = this->GetAtCoordinate(rData.Density,rData.N);
+    const array_1d<double,3> body_force = density * this->GetAtCoordinate(rData.BodyForce,rData.N);
+
+    const array_1d<double,3> convective_velocity = this->FullConvectiveVelocity(rData);
+
+    double tau_one;
+    double tau_two;
+    double tau_p;
+    this->CalculateStabilizationParameters(rData,convective_velocity,tau_one,tau_two,tau_p);
+
+    const double dt = rData.DeltaTime;
+    DenseVector< array_1d<double,Dim> > mOldSubscaleVelocity = DVMS<TElementData>::mOldSubscaleVelocity;
+    // small scale velocity contributions (subscale tracking)
+    array_1d<double,Dim> OldUssTerm = (density/dt) * mOldSubscaleVelocity[rData.IntegrationPointIndex]; // rho * u_ss^{n-1}/dt
+
+    // Old mass residual for dynamic pressure subscale: -Div(u^n)
+    double OldResidual = 0.0;
+    for (unsigned int a = 0; a < NumNodes; a++)
+    {
+        const array_1d<double,3>& rOldVel = this->GetGeometry()[a].FastGetSolutionStepValue(VELOCITY,1);
+        double OldDivProj = this->GetGeometry()[a].FastGetSolutionStepValue(DIVPROJ,1);
+        for (unsigned int d = 0; d < Dim; d++)
+            OldResidual -= rData.DN_DX(a,d)*rOldVel[d] + rData.N[a]*OldDivProj;
+    }
+
+    Vector AGradN;
+    this->ConvectionOperator(AGradN,convective_velocity,rData.DN_DX);
+
+    // These two should be zero unless we are using OSS
+    const array_1d<double,3> MomentumProj = this->GetAtCoordinate(rData.MomentumProjection,rData.N);
+    const double MassProj = this->GetAtCoordinate(rData.MassProjection,rData.N);
+    const double fluid_fraction = this->GetAtCoordinate(rData.FluidFraction, rData.N);
+    const double fluid_fraction_rate = this->GetAtCoordinate(rData.FluidFractionRate, rData.N);
+    const double mass_source = this->GetAtCoordinate(rData.MassSource, rData.N);
+    array_1d<double, 3> fluid_fraction_gradient = this->GetAtCoordinate(rData.FluidFractionGradient, rData.N);
+    // Multiplying convective operator by density to have correct units
+    AGradN *= density;
+
+    // Note: Dof order is (u,v,[w,]p) for each node
+    for (unsigned int i = 0; i < NumNodes; i++) {
+
+        unsigned int row = i*BlockSize;
+
+        // LHS terms
+        for (unsigned int j = 0; j < NumNodes; j++) {
+            unsigned int col = j*BlockSize;
+
+            // Some terms are the same for all velocity components, calculate them once for each i,j
+
+            // Skew-symmetric convective term 1/2( v*grad(u)*u - grad(v) uu )
+            //double K = 0.5*(rN[i]*AGradN[j] - AGradN[i]*rN[j]);
```

Comment:
I do not remember discussing the possibility of using this skew-symmetric form. It could be something else to study in the future

Reply:
This is inherited from the `d_vms.cpp`, I prefer to keep this line commented.